backend/backend/exceptions.py

```diff
@@ -39,3 +39,15 @@ def custom_exception_handler(exc, context) -> Response:  # type: ignore
         response.data["status_code"] = response.status_code
 
     return response
+
+
+class UnstractFSException(UnstractBaseException):
+    """Handles all error from fs connector class and propagates the error
+    message to UI.
+
+    Args:
+        UnstractBaseException: Inherits class UnstractBaseException
+    """
+
+    default_detail = "Error testing connection. "
+    status_code = 500
```

Comment:
@kirtimanmishrazipstack might be okay for now, however we need to correctly represent these status codes as well to avoid unwanted 500 alerts.
For example, this can be a case of 400 also with an invalid path entered by the user. In order to effect this we might have to pass the status code also from the connector package.

cc: @hari-kuriakose